html2asketch/model/base.js

```diff
@@ -7,11 +7,17 @@ class Base {
     this._style = null;
     this._objectID = generateID();
     this._name = '';
-    this._resizingConstant = 63;
+    this.setResizingConstraints('none');
   }
 
   setFixedWidthAndHeight() {
-    this._resizingConstant = 12;
+    this.setResizingConstraints('width', 'height');
+  }
+
+  setResizingConstraints(first, ...rest) {
+    const rCV = resizingConstraintValues;
+
+    return this._resizingConstant = rest.reduce((acc, item) => acc & rCV[item], rCV[first]);
```

Comment:
Couple of things here:
1. `_resizing**Constant**` vs `setResizing**Constraints**` - IMO we should make it "constraints" everywhere, WDYT?
1. we don't need to return the value from the setter IMO
1. logic that adds the values (`.reduce`) should be in utils so it's easier to test (and it'd be nice to have a test 😉 )
1. if value is not found we will be doing `52 & undefined => 0`, maybe we should throw in such case? WDYT?
1. There are some invalid combinations of constraints e.g. Sketch doesn't allow you to set 'width' when you set 'left' and 'right' (same with 'height' and 'top' + 'bottom')

<img width="226" alt="screen shot 2018-04-20 at 11 28 56" src="https://user-images.githubusercontent.com/985504/39044853-a4c373b4-4491-11e8-90e5-62e73ace5a76.png">

And w/o checks we will get `59 & 62 & 61 => 56` which may confuse Sketch. I think we should have some check here and throw in these two cases.

----

I'd aim to have something like that here:
```javascript
import {calculateResizingConstraintValue} form 'utils';

setResizingConstraints(constraints) {
  this._resizingConstraint = calculateResizingConstraintValue(constraints);
}
```

Reply:
Thanks for the feedback

> 1. _resizing**Constant** vs setResizing**Constraints** - IMO we should make it "constraints" everywhere, WDYT?

Yep agree I kept it as Constant as that is what you had already in the codebase.

> 3. logic that adds the values (.reduce) should be in utils so it's easier to test (and it'd be nice to have a test 😉 )

Indeed, knew I forgot tests somewhere!

> 4. if value is not found we will be doing 52 & undefined => 0, maybe we should throw in such case? WDYT?

Yep makes sense perhaps given an undefined argument it should just return `63` which is `none`

> 5. There are some invalid combinations of constraints e.g. Sketch doesn't allow you to set 'width' when you set 'left' and 'right' (same with 'height' and 'top' + 'bottom')

Oh didn't think about that will investigate and see what sketch does in those cases.